File: widget/lib_common/src/main/java/com/yyxnb/common/action/CommonAction.java
package com.yyxnb.common.action;

import android.app.Activity;
import android.content.Context;
import android.content.ContextWrapper;

import com.yyxnb.common.CommonManager;

/**
 * 通用意图
 */
public interface CommonAction {

    /**
     * 获取 Context
     */
    Context getContext();

    /**
     * 获取 Activity
     */
    default Activity getActivity() {
        Context context = getContext();
        do {
            if (context instanceof Activity) {
                return (Activity) context;
            } else if (context instanceof ContextWrapper) {
                context = ((ContextWrapper) context).getBaseContext();
            } else {
                return null;
            }
        } while (context != null);
        return null;
    }

    /**
     * toast
     *
     * @param text
     */
    default void toast(CharSequence text) {
        CommonManager.getInstance().toast(text.toString());
    }

    /**
     * log
     *
     * @param tag
     * @param text
     */
    default void log(String tag, CharSequence text) {
        CommonManager.getInstance().log(tag, text.toString());
    }

    /**
     * log
     *
     * @param text
     */
    default void log(CharSequence text) {
        CommonManager.getInstance().log(text.toString());
    }
}

File: widget/lib_common/src/main/java/com/yyxnb/common/action/CommonAction.kt
package com.yyxnb.common.action

import com.yyxnb.common.CommonManager
import com.yyxnb.common.utils.log.LogUtils
import com.yyxnb.common.utils.log.LogUtils.e
import com.yyxnb.widget.action.WidgetAction

/**
 * 通用意图
 *
 * @author yyx
 */
interface CommonAction : WidgetAction {
    /**
     * toast
     *
     * @param text
     */
    fun toast(text: CharSequence) {
        CommonManager.toast(text.toString())
    }

    /**
     * log
     *
     * @param tag
     * @param text
     */
    fun log(tag: String, text: CharSequence) {
        CommonManager.log(tag, text.toString())
    }

    /**
     * log
     *
     * @param text
     */
    fun log(text: CharSequence) {
        CommonManager.log(text.toString())
    }

    /**
     * log e
     *
     * @param text
     */
    fun loge(text: CharSequence) {
        e(text.toString())
    }

    /**
     * log list
     *
     * @param list 集合
     */
    fun list(list: List<*>?) {
        LogUtils.list(list)
    }
}